### PR TITLE
Remove some uses of mem::uninitialized

### PR DESCRIPTION
**Pull request by tesuji**
cc #62397
r? @RalfJung

### CODE REVIEW

File: src/libterm/win.rs [hunk 1]
```diff
@@ -20,19 +20,36 @@ pub struct WinConsole<T> {
...
 struct CONSOLE_SCREEN_BUFFER_INFO {
-    dwSize: [libc::c_short; 2],
-    dwCursorPosition: [libc::c_short; 2],
+    dwSize: COORD,
+    dwCursorPosition: COORD,
     wAttributes: WORD,
-    srWindow: [libc::c_short; 4],
-    dwMaximumWindowSize: [libc::c_short; 2],
+    srWindow: SMALL_RECT,
+    dwMaximumWindowSize: COORD,
```
Comment (RalfJung) [inline src/libterm/win.rs]: What is all of this diff about?

Comment (lzutao) [reply src/libterm/win.rs]: Just to make `CONSOLE_SCREEN_BUFFER_INFO` definition like the one in winapi.

Comment (RalfJung) [reply src/libterm/win.rs]: I don't think I can review Windows FFI code I am afraid. I can't even remember how big a `DWORD` is.

**Comment by RalfJung** [comment]
r? @alexcrichton 
The `MaybeUninit` parts look good to me, but there's also some Windows FFI changes.

**Comment by lzutao** [comment]
Rebased to tidy history.

**Pull request merged**